google-cloud-trace/lib/google/cloud/trace/async_reporter.rb

```diff
@@ -22,101 +24,326 @@ module Trace
       # # AsyncReporter
       #
       # @private Used by the {Google::Cloud::Trace::Middleware} to
-      # asynchronously update traces to Stackdriver Trace service when used in
-      # a Rack-based application.
+      # asynchronously buffer traces and push batches to Stackdriver Trace
+      # service when used in a Rack-based application.
       class AsyncReporter
-        include Stackdriver::Core::AsyncActor
+        include MonitorMixin
 
         ##
-        # @private Default Maximum blacklog size for the job queue
-        DEFAULT_MAX_QUEUE_SIZE = 1000
+        # @private Implementation accessors
+        attr_reader :service, :max_bytes, :max_count, :max_queue, :interval,
+                    :threads
 
         ##
-        # The {Google::Cloud::Trace::Service} object to patch traces
-        attr_accessor :service
+        # @private Creates a new AsyncReporter instance.
+        def initialize service, max_count: 1000, max_bytes: 4000000,
+                       max_queue: 100, interval: 5, threads: 10
+          @service = service
 
-        ##
-        # The max number of items the queue holds
-        attr_accessor :max_queue_size
+          @max_count = max_count
```

Comment:
Are these values something that should be able to be specified by the user? Perhaps in `Google::Cloud::Trace.configure`?

Reply:
Potentially. But again, let's punt on that given that we want to consolidate on opencensus's instrumentation code.